Objects/bytesobject.c

```diff
@@ -1597,14 +1615,13 @@ static Py_hash_t
 bytes_hash(PyObject *self)
 {
     PyBytesObject *a = _PyBytes_CAST(self);
-_Py_COMP_DIAG_PUSH
-_Py_COMP_DIAG_IGNORE_DEPR_DECLS
-    if (a->ob_shash == -1) {
+    Py_hash_t hash = get_ob_shash(a);
+    if (hash == -1) {
         /* Can't fail */
-        a->ob_shash = Py_HashBuffer(a->ob_sval, Py_SIZE(a));
+        hash = Py_HashBuffer(a->ob_sval, Py_SIZE(a));
+        set_ob_shash(a, hash);
     }
-    return a->ob_shash;
-_Py_COMP_DIAG_POP
+    return get_ob_shash(a);
```

Comment:
You can just return hash here directly.
```suggestion
    return hash;
```

Reply:
> This should be `get_ob_shash()` again to catch when the value was just set.

Going off @ZeroIntensity review here. I think using the getter here makes more sense.

Reply:
I think returning `hash` like @kumaraditya303 suggests makes sense here.

> My understanding for using the get_ob_shash(a) is that the getter ensures ob_shash isn't changed while it is being read.

Not exactly. Atomics loads with `memory_order_relaxed` that we use here basically achieve two things:

1) The load is atomic (i.e., a single indivisible operation). In particular, the compiler is not going to load parts of the field separately and then combine it. (In practice, that would be unlikely to happen anyways.)
2) The main thing is that it avoids undefined behavior. The language declares that data races, such as concurrent reads and writes, are undefined behavior. However, concurrent reads and writes that use atomics are not undefined behavior. So that means that thread sanitizer won't issue warnings here.

Reply:
Thanks, Appreciate the detailed explanation! 

Question isn't the return non atomic though? Is that case we should worry about?

Reply:
You don't need to worry about the `return`. Atomics only matter for loads and stores -- things like reading and writing variables. They're not applicable to the `return` keyword.

Reply:
Ah yeah, I missed the `hash` reassignment in my original review, my bad. I thought that `hash` was loaded atomically, checked for `-1`, and then *only* the underlying `ob_shash` was atomically stored with a new value, leaving `hash` with the old value. Kumar's approach is more correct.

Reply:
Thanks, for the explanations. Appreciate it